tools/pipelines/templates/upload-dev-manifest.yml

```diff
@@ -5,48 +5,113 @@ parameters:
 - name: buildDirectory
   type: string
 
-- name: STORAGE_ACCOUNT
+- name: buildToolsVersionToInstall
   type: string
+  default: repo
 
-- name: STORAGE_KEY
+# The path to the pnpm store.
+- name: pnpmStorePath
   type: string
+  default: $(Pipeline.Workspace)/.pnpm-store
 
-steps:
-- task: Bash@3
-  displayName: Generate release reports
-  inputs:
-    targetType: 'inline'
-    workingDirectory: ${{ parameters.buildDirectory }}
-    script: |
-      mkdir generate_release_reports
-      flub release report -g client -o generate_release_reports --baseFileName manifest
-
-- task: Bash@3
-  displayName: Update release report version
-  inputs:
-    targetType: 'inline'
-    workingDirectory: ${{ parameters.buildDirectory }}
-    script: |
-      mkdir upload_release_reports
-      flub release report-unreleased --version $(SetVersion.version) --fullReportFilePath generate_release_reports/manifest.full.json --outDir upload_release_reports --branchName '$(Build.SourceBranch)'
-
-- task: CopyFiles@2
-  displayName: Copy release reports
-  inputs:
-    SourceFolder: ${{ parameters.buildDirectory }}/upload_release_reports
-    TargetFolder: $(Build.ArtifactStagingDirectory)/release_reports
-
-- task: AzureCLI@2
-  displayName: Upload release reports
-  continueOnError: true
-  inputs:
-    azureSubscription: 'fluid-docs'
-    scriptType: bash
-    workingDirectory: ${{ parameters.buildDirectory }}
-    scriptLocation: inlineScript
-    inlineScript: |
-      for file in upload_release_reports/*; do
-          az storage blob upload -f "$file" -c 'manifest-files' -n "$(basename "$file")" --account-name ${{ parameters.STORAGE_ACCOUNT }} --account-key ${{ parameters.STORAGE_KEY }} --overwrite true --verbose
-      done
-      # Delete generate_release_reports and upload_release_reports folder
-      rm -r generate_release_reports upload_release_reports
+jobs:
+  - job:
+    displayName: Upload Release Reports
+    variables:
+    - group: storage-vars
+    - name: version
+      value: $[stageDependencies.build.build.outputs['SetVersion.version']]
+    steps:
+    # These steps should ONLY run if we're using the repo version of the build tools. These steps are mutually exclusive
+    # with the next group of steps.
+    - ${{ if eq(parameters.buildToolsVersionToInstall, 'repo') }}:
```

Comment:
In my demo changes in https://github.com/microsoft/FluidFramework/pull/21018 I pulled the "install build-tools" step into its own template so it could be shared because indeed, it is helpful to have them available in different parts of the pipeline and we want it centralized for maintenance. Anyway that PR has other stuff but I was planning to break pieces of it out into separate PRs. But I won't get to this one for awhile, so @sonalideshpandemsft consider pulling the install-build-tools stuff from that PR and opening a new one. I can review along with Alejandro.